src/Controls/DateTime/AbstractDateTimeInput.php

```diff
@@ -0,0 +1,380 @@
+<?php declare(strict_types = 1);
+
+namespace Contributte\Forms\Controls\DateTime;
+
+use Contributte\Forms\Exception\InvalidArgumentException;
+use DateTime;
+use DateTimeImmutable;
+use DateTimeInterface;
+use Nette\Forms\Controls\BaseControl;
+use Nette\Forms\Form;
+use Nette\Utils\Arrays;
+use Nette\Utils\Html;
+use function class_exists;
+use function count;
+use function in_array;
+use function is_callable;
+use function is_string;
+
+abstract class AbstractDateTimeInput extends BaseControl
+{
+
+	/** @var DateTimeParser */
+	protected $parser;
+
+	/** @var callable(DateTimeImmutable ): mixed|null */
+	protected $valueTransformation = null;
+
+	/** @var ?DateTimeInterface */
+	protected $min = null;
+
+	/** @var ?DateTimeInterface */
+	protected $max = null;
+
+	/** @var string */
+	protected $defaultHumanFormat = 'Y-m-d H:i';
+
+	/** @var string */
+	protected $htmlFormat = 'Y-m-d\TH:i';
+
+	/** @var string */
+	protected $htmlClass = '';
```

Comment:
what is this property for? One can use control prototype to set classes.

Reply:
It is meant to easier change of default html class in child classes because it is common to customize it. It is also used to set different default classes in Date/Time/DateTime